docs/docs/tutorial/creating-a-source-plugin/part-2/index.mdx

```diff
@@ -0,0 +1,682 @@
+
+```ts:title=plugin/src/gatsby-node.ts
+export type { IPluginOptions } from "./types"
+
+export { onPluginInit } from "./on-plugin-init"
+// highlight-next-line
+export { sourceNodes } from "./source-nodes"
+```
+
+In a second terminal window run (or restart if it's still running) the `develop:site` script:
+
+```shell
+yarn develop:site
+```
+
+You should see the new log `info Example plugin sourceNodes...` printed to the terminal.
+
+### Fetch data
+
+After learning how to add the `sourceNodes` API to your plugin, it's time to do more than just outputting a message.
+
+Open the `source-nodes.ts` file, remove the previous `reporter` usage, and import the `fetchGraphQL` utility. You'll use it to make requests against the example GraphQL API and it returns an object containing `data` and `errors` (you'll only use `data` for now).
+
+```ts:title=plugin/src/source-nodes.ts
+import type { GatsbyNode } from "gatsby"
+import { fetchGraphQL } from "./utils"
+
+export const sourceNodes: GatsbyNode[`sourceNodes`] = async (gatsbyApi) => {
+  const { data } = await fetchGraphQL(
+    `http://localhost:4000/graphql`,
+    ``
+  )
+}
+```
+
+The first argument to `fetchGraphQL` is the endpoint, the second argument is the GraphQL query you want to make against the API.
+
+For this tutorial you'll be querying the `Post` and `Author` entities from the example API. You can see the query for that in the next step.
+
+<Collapsible
+  summary={<em>Working with a remote backend API</em>}
+>
+
+The topic of "working with a remote backend API" is a really broad topic and not something we can cover here in its entirety as part of this tutorial. However, here are some comments and tips that will help you while working on your plugin:
+
+- Feel free to either use a barebones fetch function (via [`node-fetch`](https://www.npmjs.com/package/node-fetch)) or something more batteries-included like [`got`](https://www.npmjs.com/package/got). It depends on your API, e.g. if it's unreliable, add retry logic to your fetching. If it's your small API that you know, a more barebones approach will probably be just fine.
+- Check if the source you're trying to access has its own SDK. Many CMSs have their own SDK, e.g. Contentful with their [`contentful`](https://www.npmjs.com/package/contentful) package. When available, we'd recommend using these first.
+- It's good practice to gracefully handle errors when trying to access a remote API. This includes errors directly from the API but also ones like sudden internet connection loss. You don't want to silently fail on such errors but show them to the user of your plugin.
+
+You can also explore the API if you're not familiar with it or have it written yourself. For example, you can visit `http://localhost:4000/graphql` to explore the example API of this tutorial. It's a fully functional GraphiQL IDE window and with its help you can create your queries and test them.
```

Comment:
Does this mean write it yourself instead? It was a little unclear to me reading this sentence.
> or have it written yourself

Reply:
What I wanted to convey:

You can use the GraphiQL IDE to explore the remote API to get to know its shape. This is helpful when you're not familiar with the API. Or if you haven't authored the API yourself (because then I'm assuming you're familiar with the API).

But writing it out like this now I think the latter part is a duplicate, and just delete the "or have written it yourself." portion.